src/repositories/updater.rs

```diff
@@ -0,0 +1,366 @@
+use crate::error::Result;
+use crate::repositories::{GitHub, GitLab};
+use crate::utils::MetadataPackage;
+use crate::{db::Pool, Config};
+use chrono::{DateTime, Utc};
+use log::{debug, info, trace, warn};
+use once_cell::sync::Lazy;
+use postgres::Client;
+use regex::Regex;
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+
+pub trait RepositoryForge {
+    /// Result used both as the `host` column in the DB and to match repository URLs during
+    /// backfill.
+    fn host(&self) -> &str;
+
+    /// FontAwesome icon used in the front-end.
+    fn icon(&self) -> &'static str;
```

Comment:
Maybe this should just be an associated constant instead?

Reply:
Apparently not:

```
error[E0038]: the trait `RepositoryForge` cannot be made into an object
  --> src/repositories/updater.rs:53:15
   |
53 |     updaters: Vec<Box<dyn RepositoryForge + Send + Sync>>,
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `RepositoryForge` cannot be made into an object
   |
   = help: consider moving `ICON` to another trait
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
  --> src/repositories/updater.rs:15:11
   |
13 | pub trait RepositoryForge {
   |           --------------- this trait cannot be made into an object...
14 |     /// FontAwesome icon used in the front-end.
15 |     const ICON: &'static str;
   |           ^^^^ ...because it contains this associated `const`

error: aborting due to previous error
```